src/main/java/homework7/task1/Pair.java

```diff
@@ -0,0 +1,68 @@
+package homework7.task1;
+
+class Pair <K, V> {
+
+    private K key;
+    private V value;
+
+    private Pair(K key, V value) {
+        this.key = key;
+        this.value = value;
+    }
+
+    private Pair() {
+        this.key = null;
+        this.value = null;
+    }
+
+    private static <K, V> Pair <K, V> of(K key, V value) {
+        if (key != null && value != null) {
```

Comment:
it is unnecessary because by default you will have key = null and value = null
make like this:
private static <K, V> Pair of(K key, V value){
return new Pair<>(key, value);
}